controlplane/kubeadm/internal/cluster_test.go

```diff
@@ -101,7 +103,13 @@ func TestGetWorkloadCluster(t *testing.T) {
 	delete(emptyKeyEtcdSecret.Data, secret.TLSKeyDataName)
 	badCrtEtcdSecret := etcdSecret.DeepCopy()
 	badCrtEtcdSecret.Data[secret.TLSCrtDataName] = []byte("bad cert")
-
+	tracker, err := remote.NewClusterCacheTracker(
+		log.Log,
+		testEnv.Manager,
+	)
+	if err != nil {
+		panic(fmt.Sprintf("unable to create cluster cache tracker: %v", err))
+	}
```

Comment:
Based on the rest of this test case (eg. line 89), I think we should use gomega for the assertion here
```suggestion
	g.Expect(err).ToNot(HaveOccurred())
```